compiler/test/compilable/isZeroInit.d

```diff
@@ -78,3 +78,36 @@ static if (is(Vector!(int[4])))
     static assert(__traits(isZeroInit, Holder!(Vector!(int[4]), 0)));
     static assert(!__traits(isZeroInit, Holder!(Vector!(int[4]), 1)));
 }
+
+struct S6 {
+    union {
+        int i1;
+        float f1;
+    }
+}
+static assert(__traits(isZeroInit, S6));
+
+struct S7
+{
+    union {
+        float f2;
+        int i2;
+    }
```

Comment:
I guess the current problem (CI errors) would be uncovered by adding a regular zero-initialized field to `S7` here, before the anonymous union. That shows that you cannot simply break out of the loop if the 2nd field happens to be overlapped.

Reply:
Yes. There doesn't seem to be a way to detect anonymous unions easily at the semantic stage. So this no longer fixes anonymous unions but does fix named unions.

Reply:
P.S. It was weird that the CI didn't seem to be reporting clear errors about what had broken.

Reply:
> There doesn't seem to be a way to detect anonymous unions easily at the semantic stage.

Perhaps checking whether the byte offset is the same as the previous field - *and* whether the field is overlapped, due to bitfields.

Reply:
> Yes. There doesn't seem to be a way to detect anonymous unions easily at the semantic stage. So this no longer fixes anonymous unions but does fix named unions.

FYI `fields` is a flattened array of all found field vars, but `members` is the full parse tree. There's an extra cost to traverse recursively over all members though.

Reply:
Thanks both for the info. Perhaps I will look at fixing anonymous unions in another pull.

> and that the previous field isn't empty (T[0]).

I didn't realize that, taking account of it has fixed this pull.

Reply:
> fixing anonymous unions

Now done!